components/core/src/Utils.cpp

```diff
@@ -213,3 +219,124 @@ ErrorCode read_list_of_paths (const string& list_path, vector<string>& paths) {
 
     return ErrorCode_Success;
 }
+
+void load_lexer_from_file (std::string schema_file_path,
+                           bool reverse,
+                           log_surgeon::lexers::ByteLexer& lexer) {
+    FileReader schema_reader;
+    schema_reader.try_open(schema_file_path);
+    /// TODO: this wrapper is repeated a lot
+    log_surgeon::Reader reader_wrapper{[&](char* buf, size_t count, size_t& read_to) -> log_surgeon::ErrorCode {
+        schema_reader.read(buf, count, read_to);
+        if (read_to == 0) {
+            return log_surgeon::ErrorCode::EndOfFile;
+        }
+        return log_surgeon::ErrorCode::Success;
+    }};
+    log_surgeon::SchemaParser sp;
+    std::unique_ptr<log_surgeon::SchemaAST> schema_ast = sp.generate_schema_ast(reader_wrapper);
+    auto* delimiters_ptr = dynamic_cast<log_surgeon::DelimiterStringAST*>(
+            schema_ast->m_delimiters.get());
+    if (!lexer.m_symbol_id.empty()) {
+        throw std::runtime_error("Error: symbol_ids initialized before setting enum symbol_ids");
+    }
+    /// TODO: this is a copy of other code
```

Comment:
say where it is duplicated from and why

Reply:
Duplicated from log_surgeon, added todo to create SearchParser in log_surgeon to remove duplication